scripts/docker/zeppelin-interpreter/Dockerfile_base

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM ubuntu:16.04
+MAINTAINER Apache Software Foundation <dev@zeppelin.apache.org>
+
+ENV Z_VERSION="0.9.0-SNAPSHOT"
+ENV LOG_TAG="[ZEPPELIN_${Z_VERSION}]:" \
+    Z_HOME="/zeppelin" \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8
+
+RUN echo "$LOG_TAG update and install basic packages" && \
+    apt-get -y update && \
+    apt-get install -y locales && \
+    locale-gen $LANG && \
+    apt-get install -y software-properties-common && \
+    apt -y autoclean && \
+    apt -y dist-upgrade && \
+    apt-get install -y build-essential
+
```

Comment:
Because `spark`, `shell`, `hdfs` need Kerberos certification, So need to add

`RUN install -yq curl unzip wget grep sed vim krb5-user libpam-krb5`

Reply:
tools like `grep sed vim build-essential` are necessary ? if not, it should removed to minimze image size.

Reply:
Files of a running container should not be changed via `vim`. We should do it without `vim`, because it's has a lot of dependencies to other packages.
```bash
# Install vim on a clean ubuntu18.04
root@ac910f3d8499:/# apt install vim --no-install-recommends
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libexpat1 libgpm2 libmpdec2 libpython3.6 libpython3.6-minimal libpython3.6-stdlib libreadline7 libsqlite3-0 libssl1.1 mime-support readline-common vim-common vim-runtime xxd
Suggested packages:
  gpm readline-doc ctags vim-doc vim-scripts
Recommended packages:
  file xz-utils
The following NEW packages will be installed:
  libexpat1 libgpm2 libmpdec2 libpython3.6 libpython3.6-minimal libpython3.6-stdlib libreadline7 libsqlite3-0 libssl1.1 mime-support readline-common vim vim-common vim-runtime xxd
0 upgraded, 15 newly installed, 0 to remove and 0 not upgraded.
Need to get 12.5 MB of archives.
After this operation, 55.8 MB of additional disk space will be used.
```

`sed` is part of ubuntu18.04
```bash
root@ac910f3d8499:/# apt install sed --no-install-recommends
Reading package lists... Done
Building dependency tree       
Reading state information... Done
sed is already the newest version (4.4-2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

Reply:
We should decide between `wget` and `curl`. I would prefer `wget` because it much smaller then curl on a clean ubuntu18.04. I've never reached a situation when creating Docker images that required a `curl`. Please keep in mind, that we should install the package `ca-certificates`.

wget vs curl
```bash
root@ac910f3d8499:/# apt install wget
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  ca-certificates libpsl5 libssl1.1 openssl publicsuffix
The following NEW packages will be installed:
  ca-certificates libpsl5 libssl1.1 openssl publicsuffix wget
0 upgraded, 6 newly installed, 0 to remove and 0 not upgraded.
Need to get 2519 kB of archives.
After this operation, 7000 kB of additional disk space will be used.
```
```
root@ac910f3d8499:/# apt install curl --no-install-recommends
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libasn1-8-heimdal libcurl4 libgssapi-krb5-2 libgssapi3-heimdal libhcrypto4-heimdal libheimbase1-heimdal libheimntlm0-heimdal libhx509-5-heimdal libk5crypto3 libkeyutils1 libkrb5-26-heimdal libkrb5-3
  libkrb5support0 libldap-2.4-2 libldap-common libnghttp2-14 libpsl5 libroken18-heimdal librtmp1 libsasl2-2 libsasl2-modules-db libsqlite3-0 libssl1.1 libwind0-heimdal
Suggested packages:
  krb5-doc krb5-user
Recommended packages:
  ca-certificates krb5-locales publicsuffix libsasl2-modules
The following NEW packages will be installed:
  curl libasn1-8-heimdal libcurl4 libgssapi-krb5-2 libgssapi3-heimdal libhcrypto4-heimdal libheimbase1-heimdal libheimntlm0-heimdal libhx509-5-heimdal libk5crypto3 libkeyutils1 libkrb5-26-heimdal libkrb5-3
  libkrb5support0 libldap-2.4-2 libldap-common libnghttp2-14 libpsl5 libroken18-heimdal librtmp1 libsasl2-2 libsasl2-modules-db libsqlite3-0 libssl1.1 libwind0-heimdal
0 upgraded, 25 newly installed, 0 to remove and 0 not upgraded.
Need to get 3910 kB of archives.
After this operation, 12.4 MB of additional disk space will be used.
```

Reply:
Also, it's possible to mount an additional container if we'll need more tools for debugging...